src/pages/Marketplace/index.tsx

```diff
@@ -1,9 +1,16 @@
+import { useSearchParams } from "react-router-dom";
+import Profile from "../Profile";
 import ActiveFilters from "./ActiveFilters";
 import Cards from "./Cards";
 import Hero from "./Hero";
 import Toolbar from "./Toolbar";
 
 export default function Marketplace() {
+  const [params] = useSearchParams();
+  const slug = params.get("slug");
+
+  if (slug) return <Profile slug={slug} />;
```

Comment:
`marketplace?params..` is the same route as `marketplace` so distinguishing this needs to be inside Marketplace where search params can be parsed